src/cassette.jl

```diff
@@ -218,25 +217,48 @@ takes the collector object and returns a metagraph
 """
 function buildgraph(g,collector)
     try
-        add_vertex!(g,:name,collector.frame.args)
+        g[collector.frame.args,:label]
     catch
-        nothing
+        add_vertex!(g,:label,collector.frame.args)
     end
-    try
-        add_vertex!(g,:name,collector.frame.ret)
+
+    try 
+        g[collector.frame.ret,:label]
     catch
-        nothing
+        add_vertex!(g,:label,collector.frame.ret)
     end
-    try
-        add_edge!(g,g[collector.frame.args,:name],g[collector.frame.ret,:name],:name,collector.frame.func)
-    catch
-        nothing
+    
+    if !has_edge(g,g[collector.frame.args,:label],g[collector.frame.ret,:label])    
+        add_edge!(g,g[collector.frame.args,:label],g[collector.frame.ret,:label],:label,collector.frame.func)
     end
+                    
     for frame in collector.data
         buildgraph(g,frame)
     end
+                    
+    return g
+end
+
+"""    cleangraph(g::MetaDiGraph)
+
+removes the nothings to replace them with strings for outputing an image
+"""
+function clearngraph(mg::MetaDiGraph)
```

Comment:
these throw key errors for me. can rewrite as: 

`function cleangraph(mg::MetaDiGraph)
    for vertex in vertices(mg)
        print(vertex)
        
        try get_prop(mg, vertex, :name)
            catch e
                if isa(e, KeyError)
                set_prop!(mg,vertex,:name,"missing")
            end
        end
    end
    
    for edge in edges(mg)
        
        try get_prop(mg, edge,:name)
            catch e
                if isa(e, KeyError)
                    set_prop!(mg, edge, :name, "missing")
                end
        end        
    end
    
    return g
end`